openedx_tagging/core/tagging/models.py

```diff
@@ -195,38 +288,72 @@ def validate_object_tag(
         """
         Returns True if the given object tag is valid for the current Taxonomy.
 
-        Subclasses can override this method to perform their own validation checks, e.g. against dynamically generated
-        tag lists.
+        Subclasses should override the internal _validate* methods to perform their own validation checks, e.g. against
+        dynamically generated tag lists.
```

Comment:
It would be good to have a fully dynamic taxonomy (perhaps prime numbers or fibonacci numbers, something like that) in the test suite here. That can come in a follow-up PR though, based on learnings from the system taxonomies PR.

Reply:
Yep, I'll make sure this test case is covered by the system taxonomy PR.

Reply:
Yes, I will keep that in mind now that I make the changes in my PR